### PR TITLE
Add client secret authentication for Service Bus

### PR DESCRIPTION
**Pull request by nikhgup**
Adding support for client secret based authentication
upgrading servicebus library
adding Microsoft.Identity.client nuet package

Tracked by: https://github.com/NuGet/Engineering/issues/3650

### CODE REVIEW

**Comment by dnfgituser** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/NuGet/ServerCommon?pullRequest=359) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/NuGet/ServerCommon?pullRequest=359) before we can accept your contribution.<br/><br/>:x: nikhgup [sign now](https://cla.dotnetfoundation.org/NuGet/ServerCommon?pullRequest=359)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/NuGet/ServerCommon?pullRequest=359) it.</sub>

File: src/NuGet.Services.ServiceBus/TopicClientWrapper.cs [hunk 1]
```diff
@@ -16,6 +18,25 @@ public TopicClientWrapper(string connectionString, string path)
             _client = TopicClient.CreateFromConnectionString(connectionString, path);
         }
 
+        public TopicClientWrapper(String clientId, string clientSecret, string tenantId, string serviceBusUrl, string path)
```
Comment (loic-sharma) [inline src/NuGet.Services.ServiceBus/TopicClientWrapper.cs:21]:
```suggestion
        public TopicClientWrapper(string clientId, string clientSecret, string tenantId, string serviceBusUrl, string path)
```

File: src/NuGet.Services.ServiceBus/NuGet.Services.ServiceBus.csproj [hunk 2]
```diff
@@ -12,11 +12,12 @@
     <PackageReference Include="Microsoft.Extensions.Logging">
       <Version>2.2.0</Version>
     </PackageReference>
+    <PackageReference Include="Microsoft.Identity.Client" Version="4.25.0" />
```
Comment (loic-sharma) [inline src/NuGet.Services.ServiceBus/NuGet.Services.ServiceBus.csproj:15]: Could we follow the style of the other `PackageReference`s in this file? To be honest, I personally prefer your style but we should stay consistent with the project's style.

```suggestion
    <PackageReference Include="Microsoft.Identity.Client">
        <Version>4.25.0</Version>
    </PackageReference>
```

File: src/NuGet.Services.ServiceBus/NuGet.Services.ServiceBus.csproj [hunk 3]
```diff
@@ -12,11 +12,14 @@
     <PackageReference Include="Microsoft.Extensions.Logging">
       <Version>2.2.0</Version>
     </PackageReference>
+	<PackageReference Include="Microsoft.Identity.Client">
+		<Version>4.25.0</Version>
```
Comment (loic-sharma) [inline src/NuGet.Services.ServiceBus/NuGet.Services.ServiceBus.csproj:16]: The spacing is off here:

```suggestion
      <Version>4.25.0</Version>
```

**Review by loic-sharma — APPROVED** [review]
Your changes look good! I also tested that we can send and receive messages successfully after your Service Bus changes

File: src/NuGet.Services.ServiceBus/NuGet.Services.ServiceBus.csproj [hunk 4]
```diff
@@ -12,16 +12,19 @@
...
     </PackageReference>
+    <PackageReference Include="Microsoft.Identity.Client">
+      <Version>4.25.0</Version>
+    </PackageReference>
     <PackageReference Include="Newtonsoft.Json">
       <Version>10.0.3</Version>
     </PackageReference>
     <PackageReference Include="WindowsAzure.ServiceBus">
-      <Version>4.1.3</Version>
+      <Version>6.0.3</Version>
```
Comment (joelverhagen) [inline src/NuGet.Services.ServiceBus/NuGet.Services.ServiceBus.csproj:22]: This is a little spooky. Have we tested all of our Service Bus jobs?

File: src/NuGet.Services.ServiceBus/TopicClientWrapper.cs [hunk 5]
```diff
@@ -16,6 +18,25 @@ public TopicClientWrapper(string connectionString, string path)
             _client = TopicClient.CreateFromConnectionString(connectionString, path);
         }
 
+        public TopicClientWrapper(string clientId, string clientSecret, string tenantId, string serviceBusUrl, string path)
+        {
+            AzureActiveDirectoryTokenProvider.AuthenticationCallback authCallback = async (audience, authority, state) =>
```
Comment (joelverhagen) [inline src/NuGet.Services.ServiceBus/TopicClientWrapper.cs:23]: Beautiful. Nice work 🥇 .

**Review by joelverhagen — APPROVED** [review]

File: src/NuGet.Services.ServiceBus/NuGet.Services.ServiceBus.csproj (hunk 4 above)
Comment (loic-sharma) [reply src/NuGet.Services.ServiceBus/NuGet.Services.ServiceBus.csproj:22]: Agreed. I tested 1 receiver scenario (Orchestrator) and two sender scenarios (Gallery and Orchestrator). Our jobs all use the same underlying framework to send/receive Service Bus messages, so I think this should cover all our jobs.

**Review by lyndaidaii — APPROVED** [review]

**Pull request merged**